include/boost/unordered/detail/foa.hpp

```diff
@@ -1375,14 +1386,18 @@ table:empty_value<Hash,0>,empty_value<Pred,1>,empty_value<Allocator,2>
   template<typename... Args>
   BOOST_FORCEINLINE std::pair<iterator,bool> emplace(Args&&... args)
   {
-    using emplace_type = typename std::conditional<
+    /* We dispatch based on whether or not the value_type is constructible from
+     * an rvalue refernce to the deduced emplace_type. We do this specifically
+     * for the csae of the node-based containers. To this end, we're able to
+     * avoid allocating a node when a duplicate element is attempted to be
+     * inserted. For immovable types, we instead dispatch to the routine that
+     * unconditionally allocates via `type_policy::construct()`.
+     */
+    return emplace_dispatch(
       std::is_constructible<
-        init_type, Args...
-      >::value,
-      init_type,
-      value_type
-    >::type;
-    return emplace_impl(emplace_type(std::forward<Args>(args)...));
+        value_type,
+        emplace_type<Args...>&&>{},
```

Comment:
I'm not 100% sure on this. 

I think we're covered because `is_constructible` is required to check `std::declval<Arg>()` so we passively should have rvalue references builtin.

If we can narrow down a problematic case here, that'd be great.

Reply:
Can't narrow down (and I tried) so I guess we're good.